index.d.ts

```diff
@@ -10,226 +10,233 @@ declare global {
+    transaction: Transaction,
+    options?: SignatureOptions
+  ): Promise<Transaction>;
+
+  /**
+   * Get the permissions allowed for you site by the user.
+   *
+   * @returns Promise of a list of permissions allowed for your dApp.
+   */
+  getPermissions(): Promise<PermissionType[]>;
+
+  /**
+   * Encrypt a string, using the user's wallet.
+   *
+   * @param data Buffer-like structure to encrypt.
+   * @param options Encrypt options. See https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/encrypt#algorithm.
+   *
+   * @returns Promise of the encrypted `Uint8Array`.
+   */
+  encrypt(
+    data: BinaryDataType,
+    options?: RsaOaepParams | AesCtrParams | AesCbcParams | AesGcmParams,
+  ): Promise<Uint8Array>;
+
+  /**
+   * Decrypt a string encrypted with the user's wallet.
+   *
+   * @param data Buffer-like structure to decrypt.
+   * @param options Decrypt options. See https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/decrypt#algorithm.
+   *
+   * @returns Promise of the decrypted `Uint8Array`.
+   */
+  decrypt(
+    data: BinaryDataType,
+    options?: RsaOaepParams | AesCtrParams | AesCbcParams | AesGcmParams,
+  ): Promise<Uint8Array>;
+
+  /**
+   * Get the user's custom Arweave config set in the extension
+   *
+   * @returns Promise of the user's Arweave config
+   */
+  getArweaveConfig(): Promise<GatewayConfig>;
+
+  /**
+   * @deprecated Find alternatives at https://docs.arconnect.io/api/signature
+   */
+  signature(
+    data: BinaryDataType,
+    // https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/sign#algorithm
+    algorithm?: RsaPssParams | EcdsaParams | { name: "RSASSA-PKCS1-v1_5" } | { name: "HMAC" },
+  ): Promise<Uint8Array>;
+
+  /**
+   * Get the user's active public key, from their wallet
+   *
+   * @returns Promise of the active public key
+   */
+  getActivePublicKey(): Promise<string>;
+
+  /**
+   * Add a token to ArConnect (if it is not already present)
+   *
+   * @param id Token contract ID
+   * @param type Type of the token (asset or collectible)
+   * @param gateway Gateway config for the token
+   */
+  addToken(id: string, type?: TokenType, gateway?: GatewayConfig): Promise<void>;
+
+  /**
+   * Checks if a token has been added to ArConnect
+   * 
+   * @param id Token to check for
+   * 
+   * @returns Token added or not
+   */
+  isTokenAdded(id: string): Promise<boolean>;
+
+  /**
+   * Dispatch an Arweave transaction (preferably bundled)
+   *
+   * @param transaction Transaction to dispatch
+   * 
+   * @returns Dispatched transaction ID and type
+   */
+  dispatch(transaction: Transaction): Promise<DispatchResult>;
+
+  /**
+   * Create a deterministic secret based on the input data.
+   * 
+   * @param data Input data to generate the hash from
+   * @param options Hash configuration
+   * 
+   * @returns Hash array
+   */
+  privateHash(data: ArrayBuffer, options: SignMessageOptions): Promise<Uint8Array>; // TODO: data is ArrayBuffer or BinaryDataType
```

Comment:
Not sure why `data` here can only be `ArrayBuffer`. It seems inconsistent with the other functions.

Also, according to the example in the docs (but not the arguments in the docs), that seems to be the case (`data` can be `BinaryDataType`):

![image](https://github.com/user-attachments/assets/6bac83a3-9597-441e-a00d-b98d1a69f02c)